src/corecel/Macros.hh

```diff
@@ -226,15 +226,11 @@
     CLS& operator=(CLS&&) = delete
 
 /*!
- * \def CELER_UNUSED_UNLESS_DEVICE
+ * \def CELER_MARK_USED
```

Comment:
`CELER_MARK_USED` or `CELER_NEVER_UNUSED` I like the former (or ALWAYS_USED?) more than the double negation in "never_unused"

Reply:
I thought about "mark used" as well but I thought it would be more memorable if we have something closely parallel with the standard "maybe unused". Maybe just `CELER_MAYBE_USED` then? I thought about "always used" but it's not true... we are just marking it as "used". So maybe mark used is best but I worry that it'll be less memorable unless we have other `MARK_XX` macros

Reply:
Well, "always used" and "never unused" are equivalents so both are not true. `CELER_MAYBE_[UN]USED`then as it's effectively a replacement for the `maybe_unused` attribute.

Reply:
Maybe we should stick with your `MARK_USED` though, since it's imperative (a command-like macro) as opposed to using the decorator-like `maybe_unused`.

Reply:
Yeah, I don't have a strong opinion between these, I'd be fine with any. I agree with your point that `MARK_X` is not as memorable though. Since we have quite a few `CELER_VERB` macros, maybe `CELER_USE` or `CELER_DISCARD` since the macro is discarding the expression.

Reply:
CELER_IGNORE?

Reply:
yeah, I like "discard" because that's the nomenclature [c++ uses](https://en.cppreference.com/w/cpp/language/expressions#Discarded-value_expressions) but CELER_IGNORE sounds good as well.

Reply:
Standard uses it, discard it shall be!